enhancements/windows-containers/windows-node-egress-proxy.md

```diff
@@ -0,0 +1,351 @@
+
+N/A
+
+#### Failure Modes & Support Procedures
+
+In general the support procedures for WMCO will remain the same. There are two underlying mechanisms we rely on, the
+publishing of proxy config to cluster resources and the consuming of the published config. If either of the underlying
+mechanisms fail, Windows nodes will become proxy unaware. This could involve an issue with the user-provided proxy
+settings, the cluster network operator, OLM, or WMCO. This would result in all future egress traffic circumventing the
+proxy, which could affect inter-pod communication, existing application availability, and security. Also, the pause
+image may not be able to be fetched, preventing new Windows nodes from running workloads. This would require manual
+intervention from the cluster admin or a new release fixing whatever bug is causing the problem.
+
+## Design Details
+
+### Configuring Proxy Environment Variables
+
+As it stands today, the source of truth for cluster-wide proxy settings is the `Proxy` resource with the name `cluster`.
+The contents of the resource are both user-defined, as well as adjusted by the cluster network operator (CNO). Some
+platforms require instances to access certain endpoints to retrieve metadata for bootstrapping. CNO has logic to inject
+additional `no-proxy` entries such as `169.254.169.254` and `.${REGION}.compute.internal` into the `Proxy` resource.
+
+OLM is a subscriber to these `Proxy` settings -- it forwards the settings to the CSV of managed operators, so the WMCO
+container will automatically get the required `NO_PROXY`, `HTTP_PROXY`, and `HTTPS_PROXY` environment variables on startup.
+In fact, OLM will update and restart the operator pod with proper environment variables if the `Proxy` resource changes.
+
+On the Linux side this is achieved through `MachineConfigs` -- when global proxy settings are changed, the ignition
+config is re-rendered by MCO (for whom proxy info is injected by CVO) to include a systemd unit
+`EnvironmentFile=/etc/mco/proxy.env`, with proxy vars set in this file.
+
+WMCO will retrieve the proxy variables by watching the `rendered-worker` `MachineConfig` for changes and parse info from
+the `proxy.env` file. This could be an expensive operation, but since we already parse ignition from the `MachineConfig`
+as part of node bootstrap, the technical cost and complexity is reduced.
+
+Then, once it has the required values, the operator will set the 3 proxy environment variables on the Windows instance
+during node configuration/reconciling through Powershell.
+```powershell
+[Environment]::SetEnvironmentVariable('<name>', '<value>', 'Machine')
+```
+Note that we will set these variables at the `Machine` scope, the highest level, to ensure all WMCO-managed services
+(kubelet, containerd, etc.) will inherit them. Since Windows services read environment vars from the system registry,
+and the registry does not update until a shutdown/startup cycle, this will require rebooting the instance.
```

Comment:
> the registry does not update until a shutdown/startup cycle

This isn't true, the registry is updated when you update it. Whether or not the updated key will be used until reboot is situation dependent.